crates/net/network/src/protocol.rs

```diff
@@ -76,9 +85,80 @@ pub enum OnNotSupported {
     Disconnect,
 }
 
+/// An established rlpx sub protocol connection as returned by [`ConnectionHandler`].
+pub trait Connection<
+    StreamedType = dyn fmt::Debug,
+    SunkType = Box<dyn fmt::Debug>,
+    E = dyn error::Error,
+>: Stream<Item = StreamedType> + Sink<SunkType, Error = E> + Send + 'static where
```

Comment:
very smol nit: let's add self constraints after `where`: `where Self: Stream<Item = StreamedType> + Sink<SunkType, Error = E> + Send + 'static`. makes it a bit easier to parse (same for `P2PConnection`)